### ISSUE TITLE
User avatar/menu behavior inconsistent between browsers

### ISSUE THREAD

**Issue by erinemay** [iFixit/react-commerce#205]
When a user is logged in, the user avatar button behaves differently depending on browser.

### Expected behavior
Chrome and FF
- The user menu is displayed
- User avatar loads

https://user-images.githubusercontent.com/72166715/157723041-ea7c5275-34e2-4f53-a023-a1772b588314.mp4

### Actual behavior
MacOS/iOS - Safari 
- User avatar doesn't load
- Clicking on it redirects to user profile on cominor

https://user-images.githubusercontent.com/72166715/157723192-2e2a0b13-975e-4ae6-868a-70776c8e772c.mp4

Repro steps:

1. Log in on cominor with the same user on both devices (make sure your test user has a unique username)
2. Visit https://react-commerce.vercel.app/store/parts
3. Click on the user avatar and observe behavior

**Comment by dhmacs** [comment]
@erinemay I might have found the culprit: https://webkit.org/blog/10218/full-third-party-cookie-blocking-and-more/

TLDR: Third party cookies are blocked by default in Safari.

## Possible solution

The only solution that comes to mind is to build a proxy endpoint into Next.js so that we can make the session cookie be first party. I need to evaluate the effort, maybe it's doable with little efforts

Thoughts?

cc @sterlinghirsh @danielbeardsley

**Comment by danielbeardsley** [comment]
> TLDR: Third party cookies are blocked by default in Safari.

This won't matter in production cause we'll be on the same domain (www.ifixit.com)... or a subdomain (store.ifixit.com) or some such, right?

So, this is only a concern in development I think. We could potentially host the dev version of the app at `{branchname}.ecomm.cominor.com` or something

**Comment by dhmacs** [comment]
@danielbeardsley That's right, in production this won't be an issue.

To achieve what you're suggesting we need to use this Vercel paid feature:

<img width="780" alt="Screenshot 2022-03-11 at 09 38 06" src="https://user-images.githubusercontent.com/4640135/157832056-8ecf57b9-89e7-4caf-960f-a1912362503a.png">

**Comment by dhmacs** [comment]
@danielbeardsley How do we want to proceed with this?

**Comment by dhmacs** [comment]
Can you try to hook it up so we can test it?

**Comment by danielbeardsley** [comment]
I'm on it, what do vercel hostnames look like?

`{branch-name}.react-commerce-ifixit.vercel.app` ?  or do we just need the root: `react-commerce-ifixit.vercel.app`

**Comment by danielbeardsley** [comment]
Got it: https://react-commerce.cominor.com/store/tools !! 

But, www.cominor.com is broken for the moment, should be back up shortly.

**Comment by danielbeardsley** [comment]
Even git branches work!
https://react-commerce-git-fix-product-list-bug-ifixit.cominor.com/store/parts
